src/main/java/application/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -26,28 +26,52 @@ public Book save(Book book) {
         try {
             session = sessionFactory.openSession();
             transaction = session.beginTransaction();
-            session.save(book);
+            session.persist(book);
             transaction.commit();
             return book;
         } catch (Exception e) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new DataProcessingException("Can't insert book " + book, e);
+            throw new EntityNotFoundException("Can't insert book " + book, e);
         } finally {
             if (session != null) {
                 session.close();
             }
         }
     }
 
+    @Override
+    public Book findByTitle(String title) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery(
+                    "FROM Book as b WHERE b.title = :title", Book.class)
+                    .setParameter("title", title).getSingleResult();
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't get book by title from db: "
+                    + title, e);
+        }
+    }
+
+    @Override
+    public Book findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery(
+                            "FROM Book as b WHERE b.id = :id", Book.class)
+                    .setParameter("id", id).getSingleResult();
+        } catch (Exception e) {

```

Comment:
the same:
1. Optional.ofNullable()
2. Here use Session.get() method instead of creation query.